### ISSUE TITLE
Add data to new nodes

### ISSUE THREAD

**Issue by afomi** [opentecture/mindmapping#10]
### Story

As a User
I want to add an arbitrary number of key/values to any object
So I can describe nodes meaningfully

### Acceptance Criteria

Given a `node` exists within `map`
When I click on a `node`
And I click `Add Key/Value`
And input new text for both Key and Value
And click `Save`
Then I see the new Key/Value is persisted.

**Comment by afomi** [comment]
@theo-armour - What'd you have in mind here?

R5 supports quite a few fields. Were you thinking an arbitrary number of custom fields?

**Comment by theo-armour** [comment]
@afomi 

> Were you thinking an arbitrary number of custom fields?

Yup. Let the JSON data file have anything JSON can have. Let the viewer app do their best to read the data. Whatever the viewer cannot read or manipulate should be left untouched.

**Comment by afomi** [comment]
@theo-armour - okay, this makes sense.

What I think about is the user interface for this. Each `node` or object, supports a standard set of fields, plus a number of arbitrary fields. From a data standpoint, an arbitrary number of fields can be a little tricky, but doable.

I've also been [looking into](https://all3dp.com/3d-file-format-3d-files-3d-printer-3d-cad-vrml-stl-obj/) other 3d file formats. `.stl` seems to be an open or "neutral" format, good for lower resolutions, whereas `.dae` files are more detailed.

Seems like there may be an opportunity for a truly open format that is convertible to other formats (with respect to limitations in various formats) - something to consider later on.

---

With the above said, do you have an immediate need for this? Or is this a longer-term consideration?

**Comment by theo-armour** [comment]
@afomi 

Generally I avoid 3D file formats and try to build everything at runtime using code.

But we are certain to have a number of conversations on this topic.

So for now let's say formats are a longer term consideration

**Comment by afomi** [comment]
@theo-armour

> 
 I avoid 3D file formats and try to build everything at runtime using code.

I think this still applies to file formats, no? The runtime building of objects is just based on code not formalized as an abstract format, but likely hardcoded in the native environment's (.js / Three.js)

Definitely to discuss more.

I understand your desire to not bother with formats yet, but as we proceed, we will be be modeling and persisting data - it will be in a format, even if implicitly.

**Comment by afomi** [comment]
The feature https://github.com/opentecture/mindmapping/issues/62 feels related to this one.
